File: backend/app/core/upgrades/stat_modifiers/player_health_1.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class PlayerHealth1(UpgradeImplementation):
    id = 'player_health_1'
    name = 'Reinforced Core'
    description = '+10 max health'
    rarity = 'common'
    type = 'stat_modifier'
    cost = 2

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = 10
        appliers.modifiers.add_multiplier('player', 'maxHealth', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/stat_modifiers/player_health_1.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const PlayerHealth1Def: UpgradeDef = {
  id: "player_health_1",
  name: "Reinforced Core",
  description: "+10 max health",
  rarity: RarityID.Common,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 2,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.MaxHealth,
  value: 10,
  stackable: true,
  maxStacks: 20,
}

export class PlayerHealth1 extends Upgrade {}
